DNN Platform/Website/admin/Skins/Logo.ascx.cs

```diff
@@ -3,82 +3,182 @@
+    using Microsoft.Extensions.DependencyInjection;
+
+    /// <summary>
+    /// Logo SkinObject.
+    /// </summary>
     public partial class Logo : SkinObjectBase
     {
-        private readonly INavigationManager _navigationManager;
-
+        private readonly INavigationManager navigationManager;
+        private readonly string svgCacheKey;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Logo"/> class.
+        /// </summary>
         public Logo()
         {
-            this._navigationManager = Globals.DependencyProvider.GetRequiredService<INavigationManager>();
+            this.navigationManager = Globals.DependencyProvider.GetRequiredService<INavigationManager>();
+            this.svgCacheKey = string.Format(DataCache.PortalCacheKey, this.PortalSettings.PortalId, this.PortalSettings.CultureCode) + "LogoSvg";
         }
 
+        /// <summary>
+        /// Gets or sets the width of the border around the image.
+        /// </summary>
         public string BorderWidth { get; set; }
 
-        public string CssClass { get; set; }
+        /// <summary>
+        /// Gets or sets the css class for the image.
+        /// </summary>
+        public string CssClass { get; set; }
+
+        /// <summary>
+        /// Gets or sets the css class for the hyperlink.
+        /// </summary>
+        public string LinkCssClass { get; set; }
 
+        /// <summary>
+        /// Gets or sets a value indicating wether to inject the svg content inline instead or wrapping it in an img tag.
+        /// </summary>
+        public bool? InjectSVG { get; set; }
+
+        /// <inheritdoc/>
         protected override void OnLoad(EventArgs e)
         {
             base.OnLoad(e);
             try
-            {
-                if (!string.IsNullOrEmpty(this.BorderWidth))
-                {
-                    this.imgLogo.BorderWidth = Unit.Parse(this.BorderWidth);
-                }
-
-                if (!string.IsNullOrEmpty(this.CssClass))
-                {
-                    this.imgLogo.CssClass = this.CssClass;
-                }
-
-                bool logoVisible = false;
+            {
+                this.imgLogo.Visible = false;
+
                 if (!string.IsNullOrEmpty(this.PortalSettings.LogoFile))
                 {
+                    // default value in case we can't load an image or svg
+                    this.litLogo.Text = this.PortalSettings.PortalName;
+
                     var fileInfo = this.GetLogoFileInfo();
                     if (fileInfo != null)
-                    {
-                        string imageUrl = FileManager.Instance.GetUrl(fileInfo);
-                        if (!string.IsNullOrEmpty(imageUrl))
-                        {
-                            this.imgLogo.ImageUrl = imageUrl;
-                            logoVisible = true;
+                    {
+                        if (this.InjectSVG.GetValueOrDefault() && fileInfo.Extension == "svg")

```

Comment:
I will adjust this here to do a case insensitive comparison on the file extension

Reply:
good idea